public/component_types/indices/knn_index.ts

```diff
@@ -55,6 +55,7 @@ export class KnnIndex extends BaseComponent implements IComponent {
     this.fields = [
       {
         label: 'Index Name',
+        name: 'indexName',
```

Comment:
Even though this is for formik purpose mostly, will this get transfered to the `type` field in the use case or still undecided?

Reply:
Likely the component [type](https://github.com/opensearch-project/opensearch-ai-flow-dashboards/pull/61/files/c77d128f01685de3a2e4e62f292711c69075a0d2#diff-0d6242f3e63e8da8fa40e9a37de958f05f18c0c5accbedda53221f24c47c5d52L35) will be more of a 1:1 relationship with a use case template step. Since we don't have defined backend step types or component building blocks yet there is nothing concrete.

Reply:
got it, I think I confused myself cause I saw the type here only for the `field` itself which was `select`. Should have expanded to see it was different then the component type you linked too.